mono/metadata/monitor.c

```diff
@@ -475,7 +475,7 @@ mono_monitor_try_enter_internal (MonoObject *obj, guint32 ms, gboolean allow_int
 		mono_monitor_allocator_lock ();
 		mon = mon_new (id);
 		if (InterlockedCompareExchangePointer ((gpointer*)&obj->synchronisation, mon, NULL) == NULL) {
-			mono_gc_weak_link_add (&mon->data, obj, TRUE);
+			mon->data = (void *)(guint64)mono_gchandle_new_weakref (obj, TRUE);
```

Comment:
Good catch. Should technically be `uintptr_t`, but we already assume `sizeof (size_t) == sizeof (void *)` all over the place, so…eh.